src/PrestaShopBundle/Form/Admin/Sell/Product/OptionsType.php

```diff
@@ -0,0 +1,176 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
+use PrestaShop\PrestaShop\Core\Domain\Product\ProductSettings;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\Length;
+
+/**
+ * This form class is responsible to generate the product options form.
+ */
+class OptionsType extends TranslatorAwareType
+{
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $productVisibilityChoiceProvider;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $productConditionChoiceProvider;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $manufacturerNameByIdChoiceProvider;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param FormChoiceProviderInterface $productVisibilityChoiceProvider
+     * @param FormChoiceProviderInterface $productConditionChoiceProvider
+     * @param FormChoiceProviderInterface $manufacturerNameByIdChoiceProvider
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        FormChoiceProviderInterface $productVisibilityChoiceProvider,
+        FormChoiceProviderInterface $productConditionChoiceProvider,
+        FormChoiceProviderInterface $manufacturerNameByIdChoiceProvider
+    ) {
+        parent::__construct($translator, $locales);
+        $this->productVisibilityChoiceProvider = $productVisibilityChoiceProvider;
+        $this->productConditionChoiceProvider = $productConditionChoiceProvider;
+        $this->manufacturerNameByIdChoiceProvider = $manufacturerNameByIdChoiceProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('active', SwitchType::class, [
```

Comment:
I understand that it's here for now because it's related to `UpdateProductOptionsCommand` But we might need to move it or duplicate it Because in the current page it's an individual toggle button in the floating footer

This UX is quite convenient so we'll see how we implement it Anyway the builder can fetch the data from multiple input fields if we need to

Reply:
we can also place this input wherever we want to when we start designing, now we have no styling so its left like this to show it using form_widget

Reply:
yep for now it's perfect here